File: test/language/benchmark/for.lua

function test_sum()
	local list = {}
	for i = 0, 999999 do
	  list[i] = i
	end
	
	local sum = 0
	for k, i in pairs(list) do
	  sum = sum + i
	end
	return sum
end

for i = 1, 10 do
  local start = os.clock()
  io.write(string.format("sum: %d time: %.8f\n",test_sum(), os.clock() - start))
end

File: test/language/benchmark/for.py
from __future__ import print_function

import time

# Map "range" to an efficient range in both Python 2 and 3.
try:
    range = xrange
except NameError:
    pass
    
def test_sum():
	list = []
	for i in range(0, 1000000):
	  list.append(i)
	
	sum = 0
	for i in list:
	  sum = sum + i
	  
	return sum

for i in range(0, 10):
  start = time.process_time()
  print("sum: "+str(test_sum())+" elapsed: " + str(time.process_time() - start))


